### ISSUE TITLE
Save/Load fails locally due to lack of file system permissions

### ISSUE THREAD

**Issue by MasterOfHash** [Explv/Explvs-AIO#19]
I've tried updating the java.policy to grant permissions but no dice. Is there anything you've had to do to get that working locally?

**Comment by Explv** [comment]
Could you print out & tell me what directory it is trying to save to?

**Comment by MasterOfHash** [comment]
```
java.security.AccessControlException: access denied ("java.io.FilePermission" "/Users/boskodaria/OSBot" "read")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at org.osbot.kB.checkPermission(yg:104)
	at java.lang.SecurityManager.checkRead(SecurityManager.java:888)
	at java.io.File.isDirectory(File.java:844)
	at com.apple.laf.AquaFileView$FileInfo.<init>(AquaFileView.java:106)
	at com.apple.laf.AquaFileView.getFileInfoFor(AquaFileView.java:126)
	at com.apple.laf.AquaFileView.getName(AquaFileView.java:154)
	at javax.swing.JFileChooser.getName(JFileChooser.java:1540)
	at com.apple.laf.AquaFileChooserUI$4.getListCellRendererComponent(AquaFileChooserUI.java:1164)
	at javax.swing.plaf.basic.BasicListUI.updateLayoutState(BasicListUI.java:1361)
	at javax.swing.plaf.basic.BasicListUI.maybeUpdateLayoutState(BasicListUI.java:1311)
	at javax.swing.plaf.basic.BasicListUI$Handler.valueChanged(BasicListUI.java:2623)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:184)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:164)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:211)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:405)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:415)
	at javax.swing.DefaultListSelectionModel.setSelectionInterval(DefaultListSelectionModel.java:459)
	at javax.swing.JList.setSelectedIndex(JList.java:2216)
	at javax.swing.plaf.basic.BasicComboPopup.setListSelection(BasicComboPopup.java:1186)
	at javax.swing.plaf.basic.BasicComboPopup.access$300(BasicComboPopup.java:65)
	at javax.swing.plaf.basic.BasicComboPopup$Handler.itemStateChanged(BasicComboPopup.java:1015)
	at javax.swing.JComboBox.fireItemStateChanged(JComboBox.java:1223)
	at javax.swing.JComboBox.selectedItemChanged(JComboBox.java:1280)
	at javax.swing.JComboBox.contentsChanged(JComboBox.java:1330)
	at javax.swing.AbstractListModel.fireContentsChanged(AbstractListModel.java:118)
	at com.apple.laf.AquaFileChooserUI$DirectoryComboBoxModel.setSelectedItem(AquaFileChooserUI.java:1245)
	at com.apple.laf.AquaFileChooserUI$DirectoryComboBoxModel.addItem(AquaFileChooserUI.java:1238)
	at com.apple.laf.AquaFileChooserUI$2.propertyChange(AquaFileChooserUI.java:399)
	at java.beans.PropertyChangeSupport.fire(PropertyChangeSupport.java:335)
	at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:327)
	at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:263)
	at java.awt.Component.firePropertyChange(Component.java:8434)
	at javax.swing.JFileChooser.setCurrentDirectory(JFileChooser.java:598)
	at javax.swing.JFileChooser.<init>(JFileChooser.java:356)
	at javax.swing.JFileChooser.<init>(JFileChooser.java:311)
	at org.aio.gui.conf_man.ConfigManager.getFileChooser(ConfigManager.java:60)
	at org.aio.gui.conf_man.ConfigManager.readConfig(ConfigManager.java:48)
	at org.aio.gui.Gui$3.doInBackground(Gui.java:340)
	at org.aio.gui.Gui$3.doInBackground(Gui.java:334)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Explv** [comment]
That's strange. The file chooser tries to set the directory to `{your home dir}/OSBot/Data`. It looks like it gets blocked on the `OSBot` dir. I would check your file system permissions for that dir.

**Comment by MasterOfHash** [comment]
Hmm I tried to set them to RW for all,
`drwxrwxrwx    5 boskodaria  staff    160 Dec 25 00:50 OSBot`

But no dice.. Any other ideas?

**Comment by MasterOfHash** [comment]
I have not yet, I was writing some other scripts. I'm at the point where I need this though so I'll dig in this afternoon.

**Comment by MasterOfHash** [comment]
Absolutely no dice. I'm stumped. I've tried wiping my security policy, setting it to all permissions, custom loading the jar with the policy via command line. Nothing works.

Not sure what to do here short of copy + pasting in JSON options

**Comment by MasterOfHash** [comment]
I'm going to go ahead and write config reader/writer that writes to AWS instead of local storage. Closing this.

**Comment by RealPoke** [comment]
I am having a similar problem now, it's pretty much the same error. But I can't use the Recourse button nor the GE button, and it gives me permissions denied in console.
The thing is that the summery file doesn't even exist. The script has not created any files or folders at all.

When I compile with Intellij the artifact is about 100MB less then when I compile with Eclipse and it crashes on start with the error that no tasks are found and the UI can't be loaded.
So I am compiling with Eclipse for now.

OS: `Arch Linux`
Kernal: `x86_64 Linux 4.20.0-arch1-1-ARCH`

Log:
`[ERROR][01/11 02:57:16 AM]: Uncaught exception!
java.security.AccessControlException: access denied ("java.io.FilePermission" "/home/martin/osbot/data/explv_aio_rsbuddy_summary.json" "read")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at org.osbot.bB.checkPermission(im:19)
	at java.lang.SecurityManager.checkRead(SecurityManager.java:888)
	at java.io.File.exists(File.java:814)
	at org.aio.activities.grand_exchange.GrandExchangeHelper.getAllGEItems(GrandExchangeHelper.java:38)
	at org.aio.gui.task_panels.ResourceTaskPanel.<init>(ResourceTaskPanel.java:33)
	at org.aio.gui.task_panels.TaskPanelFactory.createTaskPanel(TaskPanelFactory.java:9)
	at org.aio.gui.Gui.addTask(Gui.java:245)
	at org.aio.gui.Gui.lambda$3(Gui.java:97)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.awt.Dialog.show(Dialog.java:1084)
	at java.awt.Component.show(Component.java:1671)
	at java.awt.Component.setVisible(Component.java:1623)
	at java.awt.Window.setVisible(Window.java:1014)
	at java.awt.Dialog.setVisible(Dialog.java:1005)
	at org.aio.gui.Gui.open(Gui.java:232)
	at org.aio.script.AIO.lambda$2(AIO.java:125)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)`

**Comment by MasterOfHash** [comment]
I kept hitting this issue. It’s something about the way the local script
runs vs the way scripts from the SDN run. I ended up moving my configs to
the cloud though and read/write them over HTTP.

Ultimately the client has a security manager that restricts reads and
writes. For some reason, it’s supposed to grant access in that OSBot
directory but doesn’t pick that directory up properly with local scripts.
